pkg/apk/auth/chainguard.go

```diff
@@ -0,0 +1,40 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+
+	"chainguard.dev/sdk/sts"
+	"google.golang.org/api/idtoken"
+)
+
+// NewChainguardIdentityAuth returns an Authenticator that authorizes
+// requests as the given assumeable identity.
+func NewChainguardIdentityAuth(identity string) Authenticator {
+	return authenticator{
+		id:  identity,
+		iss: "https://issuer.enforce.dev", // TODO: make these configurable.
+		aud: "https://apk.cgr.dev",
+	}
+}
+
+type authenticator struct {
+	id, iss, aud string
+}
+
+func (a authenticator) AddAuth(ctx context.Context, req *http.Request) error {
+	ts, err := idtoken.NewTokenSource(ctx, a.iss)
```

Comment:
This needs to check the Host matches the APK registry before it blindly adds auth